### ISSUE TITLE
Support the same licenses in Argo that H2 supports

### ISSUE THREAD

**Issue by andrewjbtw** [sul-dlss/argo#2405]
In all places where Argo allows users to select a license, support the same set of licenses that are supported in H2

This includes:

- default object rights on the APO (currently in the edit APO form)
- setting rights on a per object basis (see #2403)
- setting rights as a bulk action (see #2404)

**Comment by peetucket** [comment]
@andrewjbtw Before beginning work on this, we may need some thoughts about how to handle the disjointness between the licenses currently being set in Argo and those being set in H2.

**Comment by andrewjbtw** [comment]
hmm - we probably need to include Amy and Hannah on this. I thought Argo just had a subset of what H2 does, not a separate set.

**Comment by andrewjbtw** [comment]
It also may require input from the access side, since the codes ultimately have to be interpreted on the purl.

**Comment by peetucket** [comment]
@amyehodge and @hannahfrost  -- we are trying to normalize licenses between Argo and H2, but they seem to be somewhat disjoint lists at the moment (or at least have separate machine readable codes, which is what ends up in the object).

Argo currently uses a combo of these two lists:
https://github.com/sul-dlss/dor-services/blob/main/lib/dor/services/creative_commons_license_service.rb
https://github.com/sul-dlss/dor-services/blob/main/lib/dor/services/open_data_license_service.rb

H2 has these:
https://github.com/sul-dlss/happy-heron/blob/main/app/models/license.rb

Any thoughts on the best way forward?

**Comment by amyehodge** [comment]
@peetucket I'm not seeing the machine readable code for the H2 licenses at all. I guess I'm not 100% clear on what the question is here.

**Comment by peetucket** [comment]
In this file for H2 https://github.com/sul-dlss/happy-heron/blob/main/app/models/license.rb the machine readable codes are the first value before the => with the human readable value after this...its basically a key-value pair.

e.g. 

'PDDL-1.0' => 'PDDL-1.0 Public Domain Dedication and License'

^^^ machine code is "PDDL-1.0" ....  ^^^^ human readable is "PDDL-1.0 Public Domain Dedication and License"

If you look at the equivalent lists for Argo, they are somewhat disjoint, i.e. H2 doesn't just extend the current list in Argo.

**Comment by peetucket** [comment]
A few notes that may argue for delaying this until a bit until we are closer to getting off of Fedora :

- The list of licenses are currently in `dor-services` gem and this where validation occurs.  We need to move where this list is stored and how validation occurs.
- As long as we store licenses in the `objectRights` datastream, note that  licenses that are not a CC or ODL do not have a canonical location in the datastream.  As long as we store licenses in Fedora, this will pose a problem for any licenses that do not fit into one of these blocks since we won't know where to store them.

We could potentially add any CC or ODL licenses in dor-services gem right now that need to be selectable in the interim that won't affect current objects or the decoupling of licenses from Fedora complications.

**Comment by andrewjbtw** [comment]
It appears that we have not been sending license information with the data from H2, so H2 has effectively not been setting licenses at all.

**Comment by jcoyne** [comment]
And one problem with sending the h2 licenses to SDR is that SDR doesn't accept the h2 licenses.

**Comment by peetucket** [comment]
See analysis in sul-dlss/dor-services-app#2546

**Comment by jcoyne** [comment]
@andrewjbtw currently we can support this in Argo/DSA/Fedora.  However, PURL only supports creativeCommons or openDataCommons licenses.  Do we need to update PURL as well?

**Comment by andrewjbtw** [comment]
Yes - I've been in touch with Mark about that.

**Comment by jcoyne** [comment]
All of the DSA work for this is complete. Remaining is to update the Argo UI and update purl.

**Comment by mjgiarlo** [comment]
@jcoyne @aaron-collier Is this issue blocked by PURL updates? Or can we make the changes in the Argo UI without causing downstream problems?

**Comment by jcoyne** [comment]
We can change Argo now, but any new languages won't display in PURL.  I'm unclear whether this ticket encompasses making the changes to Purl or whether another ticket should be created for that. I'm also unclear whether doing that work is in scope or not. Need guidance from TL or Andrew.

**Comment by mjgiarlo** [comment]
@aaron-collier @andrewjbtw :point_up:

**Comment by andrewjbtw** [comment]
I'm trying to coordinate this. I think we should hold this issue, since it includes entirely new licenses, until we work out the timing. I'm hoping we can move ahead with supporting the versioned licenses for CC/ODC before this workcycle ends. Though it's entirely possible the same change will support both the versioned licenses and the new ones.

Changes to Purl will be ticketed separately, but I'm not sure if they're already encompassed by https://github.com/sul-dlss/purl/issues/472 and https://github.com/sul-dlss/purl/issues/321

**Comment by andrewjbtw** [comment]
I filed this ticket thinking it was just to expand the Argo menu list. I didn't know what I was getting into.

**Comment by aaron-collier** [comment]
Ok - per discussion above, I'm moving to backlog for now.

**Comment by andrewjbtw** [comment]
I think this ticket can be worked on now that we've resolved the back-end questions about encoding and persisting licenses.

**Comment by andrewjbtw** [comment]
There are three places where the license menu appears in Argo:

- Edit APO form
- Item and collection show page
- Bulk update of license and rights statements

**Comment by jcoyne** [comment]
The licenses to support are here: https://github.com/sul-dlss/happy-heron/blob/main/config/licenses.yml

**Comment by andrewjbtw** [comment]
To be explicit, the Argo list will end up being longer because we still have the CC 3.0 licenses and are not planning on taking them out at this point. I think Argo also has one license not on the H2 list, in addition to the CC 3.0 ones.

**Comment by mjgiarlo** [comment]
This should have been closed by https://github.com/sul-dlss/argo/pull/3188 but I messed up the issue ID in the PR. Closing manually now.
